client/src/util.ts

```diff
@@ -12,12 +15,34 @@ export function assert(cond: unknown, msg = "Assertion failed."): asserts cond {
   }
 }
 
-let memoizedCommand: string | undefined;
+export interface DenoCommandAndVersion {
+  command: string;
+  /** The version of the deno CLI or undefined if it couldn't be determined. */
+  version: semver.SemVer | undefined;
+}
 
-export async function getDenoCommand(): Promise<string> {
-  if (memoizedCommand !== undefined) {
-    return memoizedCommand;
+export async function getDenoCommandAndVersion(): Promise<
+  DenoCommandAndVersion
+> {
+  let version: semver.SemVer | undefined;
+  let command = "deno";
+  try {
+    command = await getDenoCommand();
+    const output = await execCommand(`${command} -V`);
+    const result = /[0-9]+\.[0-9]+\.[0-9]+/.exec(output);
+    if (result != null) {
+      version = new semver.SemVer(result[0]);
+    }
+  } catch (err) {
+    console.error(`Error getting deno version: ${err}`);
   }
+  return {
+    command,
+    version,
+  };
+}
+
+export async function getDenoCommand(): Promise<string> {
```

Comment:
I don't think we should memoize this as it might prevent us from getting the latest information (I removed the memoizing). This is only used when starting the language service anyway.